### PR TITLE
Contact fields name issue

### PR DESCRIPTION
**Pull request by pankaj-ag**
- Fixed a message var parser issue. 
- Added default name in case contact.fields.name does not exists

### CODE REVIEW

File: lib/glific/contacts.ex [hunk 1]
```diff
@@ -724,6 +724,20 @@ defmodule Glific.Contacts do
         fn g, list -> [g.label | list] end
       )
     )
+    ## We change the name of the contact whenever we receive  a message form the contact.
```
Comment (AkhileshNegi) [inline lib/glific/contacts.ex]: one typo

File: test/glific/flows/message_vars_parser_test.exs [hunk 2]
```diff
@@ -246,7 +246,8 @@ defmodule Glific.Flows.MessageVarParserTest do
         "category" => "@results.ar_issue_category.category",
         "photo" => "@results.ar_issue_media",
         "othersnumber" => "@results.ar_ir_otherphone",
-        "status" => "@results.ar_issues_exitmenu"
+        "status" => "@results.ar_issues_exitmenu",
+        "storedanswer" => "@results.storedanswer.DetectedResponse"
```
Comment (AkhileshNegi) [inline test/glific/flows/message_vars_parser_test.exs:250]: @pankaj-ag didn't understand why this is needed

File: test/glific/flows/message_vars_parser_test.exs [hunk 3]
```diff
@@ -311,6 +312,11 @@ defmodule Glific.Flows.MessageVarParserTest do
           "category" => "Civic ",
           "inserted_at" => "2022-02-21T13:07:11.722277Z",
           "input" => "2"
+        },
+        "storedanswer" => %{
+          "DetectedResponse" => "",
+          "OperationStatus" => "CONTINUED_ASSESSMENT",
+          "QuestionResponseID" => 1_083_000_000_113_341
```
Comment (AkhileshNegi) [inline test/glific/flows/message_vars_parser_test.exs:319]: @pankaj-ag  didn't understand why this is needed

File: test/glific/flows/message_vars_parser_test.exs (hunk 2 above)
Comment (pankaj-ag) [reply test/glific/flows/message_vars_parser_test.exs:250]: I wrote few test cases which github was showing in the code coverage reports.

File: test/glific/flows/message_vars_parser_test.exs (hunk 3 above)
Comment (pankaj-ag) [reply test/glific/flows/message_vars_parser_test.exs:319]: I wrote few test cases which github was showing in the code coverage reports.

File: lib/glific/contacts.ex (hunk 1 above)
Comment (pankaj-ag) [reply lib/glific/contacts.ex]: Fixing that now

**Pull request merged**